colossus-testkit/src/main/scala/colossus-testkit/FakeIOSystem.scala

```diff
@@ -29,19 +30,52 @@ object FakeIOSystem {
     (sys, probe)
   }
 
-  /**
-   * Returns an entirely in-thread CallbackExecutor
-   */
-  def callingThreadCallbackExecutor(implicit system: ActorSystem): CallbackExecutor = {
-    class FakeExecutor extends Actor with CallbackExecution {
-      def receive = handleCallback
+  //prevents accidentally spinning up a new executor per test
+  private val exCache: collection.mutable.Map[ActorSystem, CallbackExecutor] = collection.mutable.Map()
+
+  def testExecutor(implicit system: ActorSystem): CallbackExecutor = {
+    exCache.get(system).getOrElse {
+      val ref = system.actorOf(Props[GenericExecutor].withDispatcher("server-dispatcher"))
```

Comment:
I doubt it.  You wouldn't be able to do anything that actually required callbacks since this executor is not a real event loop with an NIO selector.